File: python/get_security_latest_dividend_record.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
security_api = intrinio_sdk.SecurityApi()
 
identifier = 'AAPL' # str | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

try:
    api_response = security_api.get_security_latest_dividend_record(identifier)
    security = api_response.security

    print("Security Summary")
    print("-------------------------------------------------")
    print("Name:       %s" % (security.name))
    print("Code:       %s" % (security.code))
    print("Company ID: %s" % (security.company_id))
    print("Ticker:     %s" % (security.ticker))
    print("")
    print("")
    print("Latest Dividend Record")
    print("-------------------------------------------------")
    print("Announcement date: %s" % (api_response.announcement_date))
    print("Record date:       %s" % (api_response.record_date))
    print("Pay date:          %s" % (api_response.pay_date))
    print("Frequency:         %s" % (api_response.frequency))
    print("Status:            %s" % (api_response.status))
    print("Forward yield:     %s" % (api_response.forward_yield))
    print("Forward rate:      %s" % (api_response.forward_rate))

except ApiException as e:
    print("Exception when calling SecurityApi->get_security_latest_dividend_record: %s\n" % e)

File: csharp/GetSecurityLatestDividendRecord.cs
using System;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
    public class GetSecurityLatestDividendRecordExample
    {
        static void Main()
        {
            Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

            var securityApi = new SecurityApi();
            var identifier = "AAPL";  // string | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

            try
            {
                DividendRecord result = securityApi.GetSecurityLatestDividendRecord(identifier);
                SecuritySummary security = result.Security;

                Console.WriteLine("Security Summary");
                Console.WriteLine("-------------------------------------------------");
                Console.WriteLine("Name:       " + security.Name);
                Console.WriteLine("Code:       " + security.Code);
                Console.WriteLine("Company ID: " + security.CompanyId);
                Console.WriteLine("Ticker:     " + security.Ticker);
                Console.WriteLine();
                Console.WriteLine();
                Console.WriteLine("Latest Dividend Record");
                Console.WriteLine("-------------------------------------------------");
                Console.WriteLine("Announcement date: " + result.AnnouncementDate);
                Console.WriteLine("Record date:       " + result.RecordDate);
                Console.WriteLine("Pay date:          " + result.PayDate);
                Console.WriteLine("Frequency:         " + result.Frequency);
                Console.WriteLine("Status:            " + result.Status);
                Console.WriteLine("Forward yield:     " + result.ForwardYield);
                Console.WriteLine("Forward rate:      " + result.ForwardRate);

            }
            catch (Exception e)
            {
                Debug.Print("Exception when calling SecurityApi.GetSecurityLatestDividendRecord: " + e.Message);
            }
        }
    }
}
